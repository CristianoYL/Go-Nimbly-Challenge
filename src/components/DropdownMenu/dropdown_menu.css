.menu-container {
  color: var(--color-primary);
  margin: 0 0;
  display: flex;
  flex-flow: column nowrap;
}

.menu-container label {
  color: var(--color-primary);
}

#operation-select {
  color: var(--color-primary);
  background-color: var(--color-accent);
  border-color: var(--color-primary);
  margin-left: 1.5rem;
}

#operation-select option{
  background-color: var(--color-accent);
}
