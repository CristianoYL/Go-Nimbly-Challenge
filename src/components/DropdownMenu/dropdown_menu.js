import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './dropdown_menu.css';

export default class DropdownMenu extends Component {
  constructor(props) {
    super(props);

    this.state = { value: 'default' };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    // console.log(event.target.value);
    this.setState({ value: event.target.value });
    this.props.onSelected(event.target.value);
  }

  render() {
    return (
      <div className={`menu-container ${this.props.className}`}>
        <label htmlFor="operation-select" target="_select">
          {this.props.hint}
          <select
            id="operation-select"
            value={this.state.value}
            onChange={this.handleChange}
          >
            <option value="default" disabled>-- select an operation --</option>
            {this.props.children}
          </select>
        </label>
      </div>
    );
  }
}

DropdownMenu.defaultProps = { className: '' };

DropdownMenu.propTypes = {
  hint: PropTypes.string.isRequired,
  onSelected: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};
