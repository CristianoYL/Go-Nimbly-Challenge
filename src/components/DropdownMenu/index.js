import DropdownMenu from './dropdown_menu';

export default DropdownMenu;
