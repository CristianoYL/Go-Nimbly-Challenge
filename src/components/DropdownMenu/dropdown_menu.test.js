import React from 'react';
import renderer from 'react-test-renderer';
import DropdownMenu from './dropdown_menu';

test('DropdownMenu without className', () => {
  const onSelected = () => {};
  const component = renderer.create(
    <DropdownMenu onSelected={onSelected} hint="test">
      <option value="simplify">Simplify</option>
    </DropdownMenu>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DropdownMenu with all props', () => {
  const onSelected = () => {};
  const component = renderer.create(
    <DropdownMenu onSelected={onSelected} hint="test hint" className="test-name">
      <option value="simplify">Simplify</option>
    </DropdownMenu>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('props.onSelected gets called', () => {
  let oldValue = false;
  const newValue = true;
  const fakeEvent = { target: { value: true } }
  const onSelected = (value) => {
    oldValue = value;
  };
  const component = renderer.create(
    <DropdownMenu onSelected={onSelected} hint="test hint" className="test-name">
      <option value="simplify">Simplify</option>
    </DropdownMenu>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.getInstance().handleChange(fakeEvent);
  expect(oldValue).toEqual(newValue);
});
