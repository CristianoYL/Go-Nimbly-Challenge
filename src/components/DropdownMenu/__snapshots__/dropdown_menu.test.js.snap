// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu with all props 1`] = `
<div
  className="menu-container test-name"
>
  <label
    htmlFor="operation-select"
    target="_select"
  >
    test hint
    <select
      id="operation-select"
      onChange={[Function]}
      value="default"
    >
      <option
        disabled={true}
        value="default"
      >
        -- select an operation --
      </option>
      <option
        value="simplify"
      >
        Simplify
      </option>
    </select>
  </label>
</div>
`;

exports[`DropdownMenu without className 1`] = `
<div
  className="menu-container "
>
  <label
    htmlFor="operation-select"
    target="_select"
  >
    test
    <select
      id="operation-select"
      onChange={[Function]}
      value="default"
    >
      <option
        disabled={true}
        value="default"
      >
        -- select an operation --
      </option>
      <option
        value="simplify"
      >
        Simplify
      </option>
    </select>
  </label>
</div>
`;

exports[`props.onSelected gets called 1`] = `
<div
  className="menu-container test-name"
>
  <label
    htmlFor="operation-select"
    target="_select"
  >
    test hint
    <select
      id="operation-select"
      onChange={[Function]}
      value="default"
    >
      <option
        disabled={true}
        value="default"
      >
        -- select an operation --
      </option>
      <option
        value="simplify"
      >
        Simplify
      </option>
    </select>
  </label>
</div>
`;
